id.go

```diff
@@ -0,0 +1,72 @@
+package jsonschema
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+)
+
+// ID represents a Schema ID type which should always be a URI.
+// See draft-bhutton-json-schema-00 section 8.2.1
+type ID string
+
+// Validate is used to check if the ID looks like a proper schema.
+// This is done by parsing the ID as a URL and checking it has all the
+// relevant parts.
+func (id ID) Validate() error {
+	u, err := url.Parse(id.String())
+	if err != nil {
+		return errors.New("invalid URL")
```

Comment:
Would it be valuable to include the actual error here?

```suggestion
		return fmt.Errorf("invalid URL: %w", err)
```

Reply:
Agreed, no harm certainly.